src/internal/nozzle/nozzle.go

```diff
@@ -312,22 +320,44 @@ func (n *Nozzle) convertEnvelopeToPoints(envelope *loggregator_v2.Envelope) []*r
 		return n.createPointsFromGauge(envelope)
 	case *loggregator_v2.Envelope_Timer:
 		n.captureGorouterHttpTimerMetricsForRollup(envelope)
-		return []*rpc.Point{}
 	case *loggregator_v2.Envelope_Counter:
-		return []*rpc.Point{n.createPointFromCounter(envelope)}
+		return n.createPointFromCounter(envelope)
 	}
 
 	return []*rpc.Point{}
 }
 
+// if enabled configuration enableEnvelopeSelector return true when envelopSelectorTags matched specified tags
+func (n *Nozzle) hasMatchedTags(tags map[string]string) bool {
+	if !n.enableEnvelopeSelector {
+		return true
+	}
+
+	for _, t := range n.envelopSelectorTags {
+		if _, hasTagKey := tags[t]; hasTagKey {
+			return true
+		}
+	}
+
+	n.metrics.Inc(metrics.NozzleSkippedEnvelopsByTagTotal)
+	return false
+}
+
 func (n *Nozzle) createPointsFromGauge(envelope *loggregator_v2.Envelope) []*rpc.Point {
+	tags := envelope.GetTags()
+	if !n.hasMatchedTags(tags) {
+		return nil
+	}
+
 	var points []*rpc.Point
 	gauge := envelope.GetGauge()
+
```

Comment:
You can have variables storing  GetSourceId() and GetTimestamp() once , and reuse , since they are common for all metrics .
Every Get function has null check in it , which is extra perforfance cost for this commonly used function.